src/bootstrap/format.rs

```diff
@@ -101,19 +107,58 @@ pub fn format(build: &Build, check: bool) {
     }
     let ignore_fmt = ignore_fmt.build().unwrap();
 
-    let rustfmt_path = build.config.initial_rustfmt.as_ref().unwrap_or_else(|| {
-        eprintln!("./x.py fmt is not supported on this channel");
-        std::process::exit(1);
+    let rustfmt_path = build
+        .config
+        .initial_rustfmt
+        .as_ref()
+        .unwrap_or_else(|| {
+            eprintln!("./x.py fmt is not supported on this channel");
+            std::process::exit(1);
+        })
+        .to_path_buf();
+    let src = build.src.clone();
+    let (tx, rx): (SyncSender<PathBuf>, _) = std::sync::mpsc::sync_channel(128);
```

Comment:
Why 128?

Reply:
No strong reason, I just avoid unbounded queues by default and a bounded one requires me to choose some capacity. 128 items are enough to immediately spawn 16 `rustfmt` processes assuming the queue is full. The loop below picks up to 8 items per iteration. If you have more than 16 hardware threads then spawning additional ones will take a few moments for `ignore::Walk` to refill the queue, which should be fine most of the time.

Reply:
Should we calculate this automatically with `num_cpus * 8`?

Reply:
I don't think it's necessary. Traversing directories should keep this filled faster than starting and _waiting_ for processes.